nixos/modules/services/mail/postfix.nix

```diff
@@ -15,6 +15,47 @@ let
   haveVirtual = cfg.virtual != "";
   haveLocalRecipients = cfg.localRecipients != null;
 
+  preparePostfixScript = pkgs.writeScript "prepare-postfix" ''
+    #!${pkgs.stdenv.shell}
+
+    set -euo pipefail
+
+    # Backwards compatibility
+    if [ ! -d /var/lib/postfix ] && [ -d /var/postfix ]; then
+      mkdir -p /var/lib
+      mv /var/postfix /var/lib/postfix
+    fi
+
+    # All permissions set according ${pkgs.postfix}/etc/postfix/postfix-files script
+    mkdir -p /var/lib/postfix /var/lib/postfix/queue/{pid,public,maildrop}
+    chmod 0755 /var/lib/postfix
+    chown root:root /var/lib/postfix
+
+    rm -rf /var/lib/postfix/conf
+    mkdir -p /var/lib/postfix/conf
+    chmod 0755 /var/lib/postfix/conf
+    ln -sf ${pkgs.postfix}/etc/postfix/postfix-files /var/lib/postfix/conf/postfix-files
+    ln -sf ${mainCfFile} /var/lib/postfix/conf/main.cf
+    ln -sf ${masterCfFile} /var/lib/postfix/conf/master.cf
+
+    ${concatStringsSep "\n" (mapAttrsToList (to: from: ''
+      ln -sf ${from} /var/lib/postfix/conf/${to}
+      ${pkgs.postfix}/bin/postalias /var/lib/postfix/conf/${to}
+    '') cfg.aliasFiles)}
+    ${concatStringsSep "\n" (mapAttrsToList (to: from: ''
+      ln -sf ${from} /var/lib/postfix/conf/${to}
+      ${pkgs.postfix}/bin/postmap /var/lib/postfix/conf/${to}
+    '') cfg.mapFiles)}
+
+    mkdir -p /var/spool/mail
+    chown root:root /var/spool/mail
+    chmod a+rwxt /var/spool/mail
+    ln -sf /var/spool/mail /var/
+
+    #Finally delegate to postfix checking remain directories in /var/lib/postfix and set permissions on them
+    ${pkgs.postfix}/bin/postfix set-permissions config_directory=/var/lib/postfix/conf
```

Comment:
The sandbox is incomplete here.  This will still run `postfix` as root, although I don't know whether there is a way around it.

Reply:
I assumed it is acceptable to have no sandbox for the prepare script, because it does not run as a network daemon.
Should put it in a sandbox as well?

To my knowledge the "main daemon" has to be run as root. It drops privileges for it's children, but there is no tight systemd integration. I am open for suggestions. :)

Reply:
I see.  In that case I'm not sure whether there is any benefit from running this in the sandbox.